boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,933 @@
+    uint16_t reg_val = 0x0000;
+
+    // Read original content of GCR Register
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2ReadRegister(SI_GCR_ADDR, &reg_val));
+
+    CLEAR_BIT(reg_val, (CSNS1_EN_MASK | CSNS0_EN_MASK));
+    SET_BIT(reg_val, (selection & (CSNS1_EN_MASK | CSNS0_EN_MASK)));
+
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2WriteRegister(SI_GCR_ADDR, reg_val));
+
+    return EXIT_CODE_OK;
+}
+
+ExitCode Io_Efuse_Aux1Aux2ConfigureEfuse(void)
+{
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2ExitFailSafeMode());
+
+    // Global config register
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2WriteRegister(SI_GCR_ADDR, GCR_CONFIG));
+
+    // Channel 0 config registers
+    EXIT_OK_OR_RETURN(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_RETRY_0_ADDR, RETRY_CONFIG));
+    EXIT_OK_OR_RETURN(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_CONFR_0_ADDR, CONFR_CONFIG));
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2WriteRegister(
+        SI_OCR_0_ADDR, OCR_LOW_CURRENT_SENSE_CONFIG));
+
+    // Channel 1 config registers
+    EXIT_OK_OR_RETURN(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_RETRY_1_ADDR, RETRY_CONFIG));
+    EXIT_OK_OR_RETURN(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_CONFR_1_ADDR, CONFR_CONFIG));
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2WriteRegister(
+        SI_OCR_1_ADDR, OCR_LOW_CURRENT_SENSE_CONFIG));
+
+    return EXIT_CODE_OK;
+}
+
+static ExitCode Io_Efuse_Aux1Aux2ExitFailSafeMode(void)
+{
+    // Set WDIN bit for next write
+    // 1_1_00000_00000_0000
+    aux1_aux2_efuse.wdin_bit_to_set = true;
+
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2WriteRegister(SI_STATR_0_ADDR, 0x0000));
+    // Disable watchdog
+    EXIT_OK_OR_RETURN(Io_Efuse_Aux1Aux2WriteRegister(SI_GCR_ADDR, GCR_CONFIG));
+
+    // Check if the the efuse is still in fail-safe mode
+    if (HAL_GPIO_ReadPin(FSOB_AUX1_AUX2_GPIO_Port, FSOB_AUX1_AUX2_Pin) ==
+        GPIO_PIN_RESET)
+    {
+        return EXIT_CODE_UNIMPLEMENTED;
+    }
+
+    return EXIT_CODE_OK;
+}
+
+static ExitCode Io_Efuse_Aux1Aux2WriteRegister(
+    uint8_t  register_address,
+    uint16_t register_value)
+{
+    return Io_Efuse_WriteRegister(
+        register_address, register_value, CSB_AUX1_AUX2_GPIO_Port,
+        CSB_AUX1_AUX2_Pin, &aux1_aux2_efuse);
+}
+
+static ExitCode Io_Efuse_Aux1Aux2ReadRegister(
+    uint8_t   register_address,
+    uint16_t *register_value)
+{
+    return Io_Efuse_ReadRegister(
+        register_address, register_value, CSB_AUX1_AUX2_GPIO_Port,
+        CSB_AUX1_AUX2_Pin, &aux1_aux2_efuse);
+}
+
+static ExitCode Io_Efuse_WriteRegister(
+    uint8_t               register_address,
+    uint16_t              register_value,
+    GPIO_TypeDef *        chip_select_port,
+    uint16_t              chip_select_pin,
+    struct Efuse_Context *e_fuse)
+{
+    uint16_t command = 0x0000U;
+
+    // Place the register address into bits 10->13
+    command = (uint16_t)(
+        command | ((register_address & EFUSE_ADDR_MASK) << EFUSE_ADDR_SHIFT));
+    // Place register value to be written into bits 0->8
```

Comment:
nit; just adding space so its more readable
```suggestion
        command | ((register_address & EFUSE_ADDR_MASK) << EFUSE_ADDR_SHIFT));
        
    // Place register value to be written into bits 0->8
```